labgrid/resource/udev.py

```diff
@@ -407,6 +407,50 @@ def path(self):
 
         return None
 
+@target_factory.reg_resource
+@attr.s(eq=False)
+class USBSDWireDevice(USBResource):
+    """The USBSDWireDevice describes an attached SDWire device,
+    it is identified via USB using udev
+    """
+
+    disk_path = attr.ib(default=None)
+    control_serial = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        self.match['ID_VENDOR_ID'] = '04e8'
+        self.match['ID_MODEL_ID'] = '6001'
+        self.match['@ID_VENDOR_ID'] = '0424'
+        self.match['@ID_MODEL_ID'] = '2640'
+        super().__attrs_post_init__()
+
+    # Overwrite the avail attribute with our internal property
+    @property
+    def avail(self):
+        return bool(self.disk_path and self.control_serial)
+
+    # Forbid the USBResource super class to set the avail property
+    @avail.setter
+    def avail(self, prop):
+        pass
+
+    # Overwrite the poll function. Only mark the SDWire as available if both
+    # paths are available.
+    def poll(self):
+        super().poll()
+        if self.device is None:
+            self.disk_path = None
+        else:
+            if not self.avail:
+                for child in self.device.parent.children:
+                    if child.subsystem == 'block' and child.device_type == 'disk':
+                        self.disk_path = child.device_node
+                self.control_serial = self.device.get('ID_SERIAL_SHORT')
```

Comment:
This leads to a warning here:
```
test_install_image.py::test_flash_image
  /usr/lib/python3.8/_collections_abc.py:660: DeprecationWarning: Will be removed in 1.0. Access properties with Device.properties.
    return self[key]

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

This should help:

```suggestion
                self.control_serial = self.device.properties.get('ID_SERIAL_SHORT')
```